### ISSUE TITLE
DDL on _ struct{} does not correctly parse when running generate

### ISSUE THREAD

**Issue by zavierboyd** [bokwoon95/sqddl#8]
Hi again,

It looks like `_` with types of `struct{}` for table ddl does not correctly parse / get added.
To reproduce:
1. Setup the empty Postgres db using the docker compose in the repo and then
2. Run `mkdir test` to use for the migrations
3. Add a file called `tables.go.txt` with the example ACTORS table 
```go
package tables

import "github.com/bokwoon95/sq"

type ACTOR struct {
	sq.TableStruct `ddl:"primarykey=actor_id"` // PRIMARY KEY (actor_id)
	ACTOR_ID       sq.NumberField
	FIRST_NAME     sq.StringField
	LAST_NAME      sq.StringField
	LATEST_FILM_ID sq.NumberField
	// CREATE UNIQUE INDEX ON actor (first_name, last_name)
	_ struct{} `ddl:"index={first_name,last_name unique}"`
	// FOREIGN KEY (latest_film_id) REFERENCES film (film_id) ON UPDATE CASCADE ON DELETE RESTRICT DEFERRABLE INITIALLY DEFERRED
	_ struct{} `ddl:"foreignkey={latest_film_id references=film.film_id onupdate=cascade ondelete=restrict deferred}"`
}
```
4. Run 
```
# Command
sqddl generate -src $POSTGRES_URL -dest tables.go.txt -output-dir test
```
```
# Output
test/20240915061125_01_schemas.sql
test/20240915061125_02_tables.sql
```
5. Check the `..._02_tables.sql` it will look something like:
```sql
CREATE TABLE actor (
    actor_id INT NOT NULL
    ,first_name TEXT
    ,last_name TEXT
    ,latest_film_id INT

    ,CONSTRAINT actor_actor_id_pkey PRIMARY KEY (actor_id)
);
```

It's not including the ddl

This has only been tested on Postgres and with sqddl version 0.4.15

---

From some extra digging it looks like the VisitStruct method on StructParser is not correctly checking for `struct{}`
https://github.com/bokwoon95/sqddl/blob/022bd489872a7b64bcd240b48d99d487f4505f73/ddl/struct_parser.go#L86
That line will match types like `string` or `bool`
If we change it to / add a check like
```go
  } else if typ, ok := astField.Type.(*ast.StructType); ok {
    structField.Type = "struct{}"
  }
```
this will match the `struct{}` type and WriteCatalog should recognise it. 

BTW thanks for the project 👍

**Comment by bokwoon95** [comment]
Thanks for the bug report, and the fix. You're right that it's not setting `"struct{}"` correctly.

**Comment by bokwoon95** [comment]
fixed by [v0.4.16](https://github.com/bokwoon95/sqddl/commit/345c14fd37f0e74ec152fc68ccb461c36bbe0019).

**Comment by zavierboyd** [comment]
Thanks for the quick fixes 👍
